gulpfile.js

```diff
@@ -297,6 +303,216 @@ gulp.task('minifyRelease', ['generateStubs'], function() {
     });
 });
 
+gulp.task('deploy', function(done) {
+    var argv = yargs.usage('Usage: delpoy -b [Bucket Name] -d [Upload Directory]')
+        .demand(['b', 'd']).argv;
+
+    var uploadDirectory = argv.d;
+    var bucketName = argv.b;
+
+    var iface = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+    });
+
+    if (argv.confirm) {
+        // skip prompt for travis
+        deployCesium(bucketName, uploadDirectory).then(function() {
+            done();
+        });
+    } else {
+        // prompt for confirmation
+        iface.question('Files from your computer will be published to the ' + bucketName + ' bucket. Continue? [y/n] ', function(answer) {
+            if (answer === 'y') {
+                deployCesium(bucketName, uploadDirectory).then(function() {
+                    done();
+                });
+            } else {
+                console.log('Deploy aborted by user.');
+                done();
+            }
+        });
+    }
+});
+
+// Deploy cesium to aws
+function deployCesium(bucketName, uploadDirectory) {
+    var readFile = Promise.promisify(fs.readFile);
+    var gzip = Promise.promisify(zlib.gzip);
+    var concurrencyLimit = 2000;
+    var cacheControl = "max-age=3600";
```

Comment:
For actual deployment, this is what we want, but for CI it doesn't make sense because changes won't be automatically reflected if the user doesn't clear their cache.  The solution is to make this an optional command line argument and when we do travis deploys we will use `no-cache` as the setting.